cmake/CompileFlags.cmake

```diff
@@ -68,14 +68,29 @@ macro(compile_flags)
             -fno-strict-aliasing
         )
 
-        add_compile_options(
-            # When we are a stable release (Release build + USE_ASSERTS not set),
-            # assertations are off, which trigger a lot of warnings. We disable
-            # these warnings for these releases.
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
-        )
+        # When we are a stable release (Release build + USE_ASSERTS not set),
+        # assertations are off, which trigger a lot of warnings. We disable
+        # these warnings for these releases.
+        if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
+                )
+        else(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-parameter>"
+            )
+        endif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+
+        if (${OPTION_FORCE_COLORED_OUTPUT})
+            if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+                add_compile_options (-fdiagnostics-color=always)
+            elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+                add_compile_options (-fcolor-diagnostics)
+            endif ()
+        endif ()
```

Comment:
Currently, we're putting the condition in the endif
(Personally I'd like to remove these, but for now, consistency)